### PR TITLE
Link Onchain Post with Discussion Post

### PR DESCRIPTION
**Pull request by niklabh**

### CODE REVIEW

File: front-end/src/components/EditablePostContent.tsx [hunk 1]
```diff
@@ -91,17 +98,117 @@ const EditablePostContent = ({ className, isEditing, isTipProposal, onchainId, p
...
 	});
 
+	const [discussionPostQuery, { data, loading }] = useDiscussionPostAndCommentsLazyQuery();
+
+	const handleLink = () => {
+		if (!linkPostId) {
+			setValidationError('Please enter a valid post id');
+		}
+
+		if (isNaN(Number(linkPostId))) {
```
Comment (aanchalgupta-ai) [inline front-end/src/components/EditablePostContent.tsx:108]: we can replace this with `if(Number.isNaN(linkPostId)` it checks for both number and NaN

Comment (SskYwaLkeR) [inline front-end/src/components/EditablePostContent.tsx:108]: We can use `if(Number.isNaN(linkPostId))` it checks for both number and NaN

File: front-end/src/components/EditablePostContent.tsx [hunk 2]
```diff
@@ -91,17 +98,117 @@ const EditablePostContent = ({ className, isEditing, isTipProposal, onchainId, p
...
+			setValidationError('Please enter a valid post id number');
+		}
+
+		try {
+			discussionPostQuery({ variables: { id: Number(linkPostId) } });
+
+			if (data && data.posts && data.posts.length) {
```
Comment (SskYwaLkeR) [inline front-end/src/components/EditablePostContent.tsx:115]: `if(data?.posts?.length)`

File: front-end/src/components/EditablePostContent.tsx [hunk 3]
```diff
@@ -91,17 +98,117 @@ const EditablePostContent = ({ className, isEditing, isTipProposal, onchainId, p
...
+
 	useEffect(() => {
 		if (isEditing) {
 			setValue('content',content);
 			setValue('title',title);
 		}
 	},[content, isEditing, setValue, title]);
 
+	useEffect(() => {
+		if (data && data.posts && data.posts.length) {
```
Comment (SskYwaLkeR) [inline front-end/src/components/EditablePostContent.tsx:152]: Same

File: front-end/src/components/EditablePostContent.tsx [hunk 4]
```diff
@@ -91,17 +98,117 @@ const EditablePostContent = ({ className, isEditing, isTipProposal, onchainId, p
...
+				setNewContent(data.posts[0].content);
+			}
+		}
+	}, [data, setValue]);
+
 	if (!author || !author.username || !content) return <div>Post content or author could not be found.</div>;
 
 	return (
 		<>
+			{showModal ?
```
Comment (SskYwaLkeR) [inline front-end/src/components/EditablePostContent.tsx:168]: Avoid ternary, we can replace it with `{showModal && <Modal />}`

**Pull request merged**